sigmastate/src/main/scala/sigmastate/SigSerializer.scala

```diff
@@ -106,9 +114,12 @@ object SigSerializer extends LazyLogging {
     *
     * @param exp   sigma proposition which defines the structure of bytes from the reader
     * @param proof proof to extract challenges from
+    * @param E     optional evaluator (can be null) which is used for profiling of operations.
```

Comment:
Will it be removed in 5.x? This parameter is pretty confusing.

Reply:
It is only necessary for profiling and setting up cost parameters of signature verification. See estimateCryptoVerifyCost method. Can be removed, but I suggest to keep it.
What exactly not clear, I can add clarifying description.

Reply:
It is not cleat at all how a serializer could be dependent on an evaluator, as the latter is higher-level entity than the former.

Reply:
@kushti Yes, agree, but the profiling infrastructure is built around evaluator. Will think how it may be refactored.